adapters/repos/db/lsmkv/cursor_segment_replace.go

```diff
@@ -101,8 +114,12 @@ func (s *segmentCursorReplace) nextWithAllKeys() (segmentReplaceNode, error) {
 		return out, lsmkv.NotFound
 	}
 
+	contents := make([]byte, s.segment.dataEndPos)
```

Comment:
entire data part of segment seems to be read here (which might be huge).
`replaceStratParseDataWithKey` can be changed to accept reader, similarly as `ParseCollectionNode`